tests/Operation/WatchFunctionalTest.php

```diff
@@ -588,6 +588,46 @@ public function testResumeTokenNotFoundAdvancesKey()
         $this->assertSame(2, $changeStream->key());
     }
 
+    public function testSessionPersistsAfterResume()
+    {
+        $operation = new Watch($this->manager, $this->getDatabaseName(), $this->getCollectionName(), [], $this->defaultOptions);
+
+        $changeStream = null;
+        $originalSession = null;
+        $sessionAfterResume = null;
+
+        // We want to ensure that the lsid of the initial aggregate matches the
+        // lsid of any aggregates after the change stream resumes. After
+        // PHPC-1152 is complete, we will ensure that the lsid of the initial
+        // aggregate matches the lsid of any subsequent aggregates and getMores.
+        (new CommandObserver)->observe(
+            function() use ($operation, &$changeStream) {
+                $changeStream = $operation->execute($this->getPrimaryServer());
+            },
+            function($changeStream) use (&$originalSession) {
+                if (isset($changeStream->aggregate)) {
+                    $originalSession = $changeStream->lsid;
+                }
+            }
+        );
+
+        $changeStream->rewind();
+        $this->killChangeStreamCursor($changeStream);
```

Comment:
Please test this against a replica set with auth if possible. That should highlight whether `killChangeStreamCursor()` will need to also take the same implicit session created within the Watch class.

Reply:
After testing with auth, it looks like `killChangeStreamCursor()` does not need to take the same implicit session created within the Watch class.

Reply:
Excellent. That answers my question in https://github.com/mongodb/mongo-php-driver/pull/803#discussion_r182100188.